test/suites/chaos/suite_test.go

```diff
@@ -0,0 +1,245 @@
+					{
+						Key:      v1alpha5.LabelCapacityType,
+						Operator: v1.NodeSelectorOpIn,
+						Values:   []string{v1alpha5.CapacityTypeSpot},
+					},
+				},
+				Consolidation: &v1alpha5.Consolidation{
+					Enabled: lo.ToPtr(true),
+				},
+				ProviderRef: &v1alpha5.ProviderRef{Name: provider.Name},
+			})
+			numPods := 1
+			dep := test.Deployment(test.DeploymentOptions{
+				Replicas: int32(numPods),
+				PodOptions: test.PodOptions{
+					ObjectMeta: metav1.ObjectMeta{
+						Labels: map[string]string{"app": "my-app"},
+					},
+					TerminationGracePeriodSeconds: lo.ToPtr[int64](0),
+				},
+			})
+			// Start a controller that adds taints to nodes after creation
+			Expect(startTaintAdder(ctx, env.Config)).To(Succeed())
+			startNodeCountMonitor(ctx, env.Client)
+
+			// Create a deployment with a single pod
+			env.ExpectCreated(provider, provisioner, dep)
+
+			// Expect that we never get over a high number of nodes
+			Consistently(func() int {
+				list := &v1.NodeList{}
+				if err := env.Client.List(env.Context, list, client.HasLabels{test.DiscoveryLabel}); err != nil {
+					return -1
+				}
+				return len(list.Items)
+			}, time.Minute*5).Should(BeNumerically("<", 35))
+		})
+		It("should not produce a runaway scale-up when ttlSecondsAfterEmpty is enabled", Label(common.NoWatch), Label(common.NoEvents), func() {
+			ctx, cancel := context.WithCancel(env.Context)
+			defer cancel()
+
+			provider := awstest.AWSNodeTemplate(v1alpha1.AWSNodeTemplateSpec{AWS: v1alpha1.AWS{
+				SecurityGroupSelector: map[string]string{v1alpha5.DiscoveryTagKey: settings.FromContext(env.Context).ClusterName},
+				SubnetSelector:        map[string]string{v1alpha5.DiscoveryTagKey: settings.FromContext(env.Context).ClusterName},
+			}})
+			provisioner := test.Provisioner(test.ProvisionerOptions{
+				Requirements: []v1.NodeSelectorRequirement{
+					{
+						Key:      v1alpha5.LabelCapacityType,
+						Operator: v1.NodeSelectorOpIn,
+						Values:   []string{v1alpha5.CapacityTypeSpot},
+					},
+				},
+				TTLSecondsAfterEmpty: lo.ToPtr[int64](30),
+				ProviderRef:          &v1alpha5.ProviderRef{Name: provider.Name},
+			})
+			numPods := 1
+			dep := test.Deployment(test.DeploymentOptions{
+				Replicas: int32(numPods),
+				PodOptions: test.PodOptions{
+					ObjectMeta: metav1.ObjectMeta{
+						Labels: map[string]string{"app": "my-app"},
+					},
+					TerminationGracePeriodSeconds: lo.ToPtr[int64](0),
+				},
+			})
+			// Start a controller that adds taints to nodes after creation
+			Expect(startTaintAdder(ctx, env.Config)).To(Succeed())
+			startNodeCountMonitor(ctx, env.Client)
+
+			// Create a deployment with a single pod
+			env.ExpectCreated(provider, provisioner, dep)
+
+			// Expect that we never get over a high number of nodes
+			Consistently(func() int {
+				list := &v1.NodeList{}
+				if err := env.Client.List(env.Context, list, client.HasLabels{test.DiscoveryLabel}); err != nil {
+					return -1
```

Comment:
Should probably return a value > 35 here so the test will fail if we can't list nodes.